docs/framework/wcf/samples/custom-lifetime.md

```diff
@@ -1,219 +1,227 @@
+
+Therefore, in order to extend the <xref:System.ServiceModel.InstanceContext>, you must implement the `IExtension` interface. In this sample project, the `CustomLeaseExtension` class contains this implementation.
+
+```csharp
+class CustomLeaseExtension : IExtension<InstanceContext>
+{
+}
+```
+
+The <xref:System.ServiceModel.IExtension%601> interface has two methods <xref:System.ServiceModel.IExtension%601.Attach%2A> and <xref:System.ServiceModel.IExtension%601.Detach%2A>. As their names imply, these two methods are called when the runtime attaches and detaches the extension to an instance of the <xref:System.ServiceModel.InstanceContext> class. In this sample, the `Attach` method is used to keep track of the <xref:System.ServiceModel.InstanceContext> object that belongs to the current instance of the extension.
+
+```csharp
+InstanceContext owner;
+
+public void Attach(InstanceContext owner)
+{
+    this.owner = owner;
+}
+```
+
+In addition, you must add the necessary implementation to the extension to provide the extended lifetime support. Therefore, the `ICustomLease` interface is declared with the desired methods and is implemented in the `CustomLeaseExtension` class.
+
+```csharp
+interface ICustomLease
+{
+    bool IsIdle { get; }
+    InstanceContextIdleCallback Callback { get; set; }
+}
+
+class CustomLeaseExtension : IExtension<InstanceContext>, ICustomLease
+{
+}
+```
+
+When WCF invokes the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle%2A> method in the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider> implementation, this call is routed to the <xref:System.ServiceModel.Dispatcher.IInstanceContextProvider.IsIdle%2A> method of the `CustomLeaseExtension`. Then, the `CustomLeaseExtension` checks its private state to see whether the <xref:System.ServiceModel.InstanceContext> is idle. If it is idle, it returns `true`. Otherwise, it starts a timer for a specified amount of extended lifetime.
+
+```csharp
+public bool IsIdle
+{
+  get
+  {
+    lock (thisLock)
+    {
+      if (isIdle)
+      {
+        return true;
+      }
+      else
+      {
+        StartTimer();
+        return false;
+      }
+    }
+  }
+}
+```
+
+In the timer’s `Elapsed` event, the callback function in the Dispatcher is called in order to start another clean-up cycle.
+
+```csharp
+void idleTimer_Elapsed(object sender, ElapsedEventArgs args)
+{
+    lock (thisLock)
+    {
+        StopTimer();
+        isIdle = true;
+        Utility.WriteMessageToConsole(
+            ResourceHelper.GetString("MsgLeaseExpired"));
+        callback(owner);
+    }
+}
+```
+
+There is no way to renew the running timer when a new message arrives for the instance being moved to the idle state.
```

Comment:
Remove this sentence as it looks to be talking about a timer in WCF which never made it to production.